src/components/source.js

```diff
@@ -114,14 +114,14 @@ function Source(props) {
         // Removing source before dependent layers will throw error
         // TODO - find a more robust solution
         requestAnimationFrame(() => {
-          if (map.style && map.style._loaded) {
+          if (map.style && map.style._loaded && map.getSource(id)) {
             map.removeSource(id);
           }
         });
       };
     }
     return undefined;
-  }, [map]);
+  }, [map, id]);
```

Comment:
`id` is not supposed to change.

Reply:
Yeah, it makes sense that it shouldn't change, but it is still a dependency of the effect, so I included it to satisfy the ["exhaustive deps" recommendation](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528)

If the component is used correctly, then adding this change will have no effect, and if the `id` does change, then this means that it is more likely that both sources are cleaned up properly.

Of course, if you prefer, I can update the PR to remove it.

Reply:
`id` is a constant ensured by the `useMemo(..., [])` call above, so it doesn't depend on whether the user is using this component correctly.

Reply:
Ah, my mistake! I should've zoomed out a bit :smile: I can update the PR no problem, up to you.

Reply:
(just an aside, `useMemo` [doesn't guarantee](https://reactjs.org/docs/hooks-reference.html#usememo) that it will be a constant)
> You may rely on useMemo as a performance optimization, not as a semantic guarantee.